pennylane/_device.py

```diff
@@ -452,6 +455,9 @@ def execute(self, queue, observables, parameters={}, **kwargs):
             # increment counter for number of executions of device
             self._num_executions += 1
 
+            if self.tracker.tracking:
+                self.tracker.update_and_record(executions=1, shots=self._shots)
```

Comment:
Seems to not have been covered, if this is the case, might be worth tracking a device that is not a subclass of `QubitDevice`: e.g., `default.gaussian`.